### ISSUE TITLE
Nushell dealbreakers from twitter

### ISSUE THREAD

**Issue by jntrnr** [nushell/nushell#3953]
### Describe the bug

From `@clementd`

"Dealbreakers were direnv support, autojump support, lack of completions, and missing equivalents for bashisms like !$"

From `@dysinger`

"Y i recently switched back to bash because nix-shell always drops me into bash anyway"

### How to reproduce

Steps above

### Expected behavior

While we have versions of direnv (like autoenv), we probably should go a bit deeper as well as have better support for existing direnv solutions.

Autojump and equiv currently need external support. I think it's coming but more there to do I think.

**Comment by clementd-fretlink** [comment]
To elaborate a bit more:

## direnv support

So there is already support for per-directory $env (https://github.com/nushell/nushell/issues/86) but
- i already have a lot of `.envrc` files that i'd like to not duplicate everywhere
- I use `nix-direnv` which allows me to spawn nix shells automatically (and quickly). I don't know of an equivalent in nushell

## autojump support (or fasd)

I could not get neither `autojump` or `fasd` to work (maybe it works now?)

Here, fish-style completion on `cd` was somehow helpful, but only for directories that don't have a common prefix.

## missing bashisms

I use `!$` _a lot_ (to run a command with the last argument of the previous one) and couldn't find something as easy to use in fish

## missing completion

Lots of CLI tools have context-sensiste autocompletion for bash. A prime example is git. There again, history-based completion helps, but is not as useful as it could. idk if it would possible to fallback on bash completion (eg. zsh does support bash completions).

I've used `nushell` for several months. I lived without autojump, `!$`, and context-sensitive completions, but in the end it was the lack of direnv support that made me switch back

**Comment by AmandaCameron** [comment]
re: direnv, I actually needed that as well, and managed to do it somewhat on my own with the following:

```nu
alias load-direnv = load-env (direnv export elvish | from json | pivot name value)
```

The main problem with it is it'll permentnatly "taint" the session, as AFAIK there's no inverse to `load-env` where we can remove the environment variables back out, so when I'm done with that directory's direnv-loaded, I just ctrl-d.

**Comment by from-nibly** [comment]
Another nix user here. Being able to use direnv specifically for nix-direnv is pretty important for me. I use that in all my repos to set up all the environment dependencies (e.g. nodejs, yarn, java whatever).

**Comment by jakeisnt** [comment]
`nix-direnv`, and consequently `direnv` support, is the only thing keeping me from switching to nushell.
Would this be possible with a plugin or would it have to be implemented as a part of nushell itself?

**Comment by alechp** [comment]
### TLDR
- I've been using nushell as my daily driver for a few months and mostly loving it
- Sometimes hit shell specific requirements that require me to change
- nushell's lack of backwards compatible with bash environment variables and file sourcing (ie. static sourced array on nushell startup vs. bash directory loop) means a degraded experience
- Going to continue using nu regardless, but hopeful that cross shell compatibility is more streamlined in future
 
### Major pain point
- Occasionally I'll run into shell incompatibility with system requirements (most recently conda) 
> ![Screen Shot 2022-01-21 at 12 24 14 PM](https://user-images.githubusercontent.com/3058939/150572201-cf409abc-15cb-498e-a63f-6b9fb18f1137.png)
- Could `chsh` back to ZSH, but I've adapted all of my bash aliases, sourcing and environment variables (hundreds/thousands of lines) to be compatible with nushell. This leads to derivative issues like "oh yeah, zplug path isn't recognized in zsh, so previous config is broken, which means bash aliases don't work" 
> <img width="1072" alt="Screen Shot 2022-01-21 at 12 28 13 PM" src="https://user-images.githubusercontent.com/3058939/150572669-e47a3ee9-df89-4263-a707-5316c72c7bd9.png">

I've tried to mostly just ignore this, but when the pain hits, it hits hard. Going to work around this by fishing out my old config from git. The issue ofc is that I have to maintain two separate shell environments. 

This isn't the first time I've had an issue like this, but I've usually been able to find a way around it (sometimes as hacky as chsh, run command I need, then chsh back; other times simply finding another tool without a specific shell dependency).

While this isn't a "dealbreaker" for me (I'm still using nu and mostly love it) it's a massive pain. And unfortunately there isn't a quickfix (that I'm aware of) like simply short-circuiting with `^conda [...]`

### Minor pain points 
- `Novelty terminals I'd like to explore` (eg. [Warp](https://www.warp.dev/), [Tabby](https://tabby.sh/)) `lack nu support`. While it makes sense (and isn't a dealbreaker) it's definitely a bummer to not be able to play around with nifty terminal options (I'm currently using [Kitty](https://sw.kovidgoyal.net/kitty/) and have no complaints, but enjoy playing around with new tech)

- `Restoration on fresh devices is a bit wonky.` I use [mackup](https://github.com/lra/mackup) (linux/mac compatible backup tool) to save all my dev configs. Because of nushell's non-standard environment variables, restoration from scratch is odd (more pathing issues/env var incompat issues)

- Whenever I need to `chsh` to work with something that isn't nu compatible, `I have to close my terminal`. I could solve this by adding tmux or saving logs, but definitely cumbersome  
- `$nu.env.VAR feels verbose` compared to $VAR
<img width="518" alt="Screen Shot 2022-01-21 at 12 44 52 PM" src="https://user-images.githubusercontent.com/3058939/150575045-69a25213-240d-44de-9f75-ea9239c68fc1.png">
- Feels odd to me that `aliases require a space` 
<img width="638" alt="Screen Shot 2022-01-21 at 12 46 16 PM" src="https://user-images.githubusercontent.com/3058939/150575240-19763c06-7806-411b-b831-55f8520daebc.png">

PS. If there are quick config fixes for any of these issues, please lmk!

**Comment by Hofer-Julian** [comment]
@alechp for conda there is https://github.com/nushell/nu_scripts/tree/main/virtual_environments
